tools/engine_tool/lib/src/environment.dart

```diff
@@ -32,6 +34,9 @@ final class Environment {
   /// Information about paths in the engine repo.
   final Engine engine;
 
+  /// Facility for commands to access files.
+  final FileSystem fileSystem;
```

Comment:
This isn't a bad idea, but all the rest of the code accesses the file system directly through `dart:io`, so adding this now without updating the fs access everywhere else will be a bit confusing.

Instead, you might think about how to update `dependenciesUpdated` and the other methods in `dependencies.dart` to be unit-testable without a fake file system. As part of this change you'd unit test that logic, but maybe wouldn't write end-to-end tests of the command lines.

Reply:
> This isn't a bad idea, but all the rest of the code accesses the file system directly through `dart:io`, so adding this now without updating the fs access everywhere else will be a bit confusing.

Would you be okay with me migrating all of `engine_tool` to this file system? Or does "the rest of the code" include dependencies like `engine_build_configs` and `engine_repo_tools` too?

Reply:
Yeah, I think it would have to include dependencies. The `Engine` object from `engine_repo_tools` vends `io.Directory` objects after being constructed by poking around the engine tree on the real fs, and the `et` environment has an `Engine` that it uses all over.

Reply:
I would be a -1 here. Not every package needs to be aware of fake file systems.

Let's stick with `dart: io`. The tests can use a real (temporary) directory, that pattern is common across our repo.

Reply:
Does the `FakeEngine` in `engine_repo_tools` do some of that already?

Reply:
> The tests can use a real (temporary) directory, that pattern is common across our repo.

+1 to Hixie's comment, using `package:file` for testing is the more common pattern in Flutter. It's a high fidelity fake & allows for faster tests that don't run into weird issues (looking at you Windows).

> If dependencies are a problem ...

That's not a problem, this tool already depends on `package:file`

Reply:
The Flutter CLI uses `package:file`, and it *does* make testing quite a bit easier than writing tests against the real local fs, even easier than creating and managing entries under /tmp.

Even with `package:file`, I suspect that a better place for reading the DEPS file, computing its hash, and storing it the dot file is in the `engine_repo_tools` package in the `Engine` object. The `TestEngine` implementation at https://github.com/flutter/engine/blob/main/tools/pkg/engine_repo_tools/lib/engine_repo_tools.dart#L172 can implement the API such that it can be used for the `et` unit tests, and the hashing code in `engine_repo_tools` can also be re-used to implement the `gclient hook`.